classes/webservice/WebserviceOutputJSON.php

```diff
@@ -163,6 +163,11 @@ public function renderNodeFooter($node_name, $params)
 
     public function overrideContent($content)
     {
+        array_walk_recursive($this->content, function (&$item) {
+            if (is_null($item)) {
+                $this->content[key($this->content)] = [];
+            }
+        });
```

Comment:
```suggestion
        array_walk_recursive($this->content, function (&$item) {
            if ($item === null) {
                $item = [];
            }
        });
```

Never use is_null method, it's slow.
I think you want to add empty array on $item, not everywhere :)

Reply:
With your solution, I get the following result :
![image](https://user-images.githubusercontent.com/22498741/105870600-bbaae780-5ff8-11eb-9c72-173776587c08.png)

I tried this, and I think I got the expected result :
```suggestion
       if(count($this->content[key($this->content)])==1){
            array_walk_recursive($this->content, function (&$item) {
                if ($item === null) {
                    $this->content[key($this->content)] = [];
                }
            });
        }
```
![image](https://user-images.githubusercontent.com/22498741/105872666-f150d000-5ffa-11eb-926b-4f6a5d00578c.png)
![image](https://user-images.githubusercontent.com/22498741/105872742-06c5fa00-5ffb-11eb-8190-7761cdf9001c.png)


Otherwise, as mentionned in [this comment](https://github.com/PrestaShop/PrestaShop/issues/22760#issuecomment-767403666), wouldn't it be easier and more relevant to just return "{}" when the query returns no results ?

Reply:
I'm not sure it's the right solution. If there is only one element, why doing an array_walk_recursive :thinking: